src/main/java/com/puppycrawl/tools/checkstyle/utils/CommonUtil.java

```diff
@@ -64,6 +64,9 @@ public final class CommonUtil {
     /** Pseudo URL protocol for loading from the class path. */
     public static final String CLASSPATH_URL_PROTOCOL = "classpath:";
 
+    /** User working directory. */
+    public static final String USER_DIR = System.getProperty("user.dir");
```

Comment:
why we need user home folder ?

if we need to create some file we can use only temp folder or any folder inside repo.
We should not pollute in any other folder at all.

Reply:
even you think some folder is required .... please move this out of Util class, we should allow any class to even try to use system properties.
If it is hack for XdocGnerator, this class is not our main feature and should be moved out of checkstyle project eventually.

Reply:
![image](https://github.com/checkstyle/checkstyle/assets/812984/d804dbb5-cc55-4ddc-bc32-2c87bcda0998)

you can not rely on fact that repo is cloned in home folder.
it is especially not actual for Windows users.
you need `XdocUtil.getXdocsTemplatesFilePaths()` to return absolute paths of files, just make new method to return what you need.

Reply:
@romani Before running the test in IDEA, i run those two and it works
```
mvn clean compile
mvn plexus-component-metadata:generate-metadata
```

Reply:
> why we need user home folder ?

For processing macro in `XdocsTemplateParser`. The parent class offers `getBasedir()` but it is deprecated
https://maven.apache.org/doxia/doxia/doxia-core/apidocs/org/apache/maven/doxia/parser/AbstractParser.html#getBasedir--

Reply:
> please move this out of Util class,

Removed from util class and usage in `generateXdocContent`. Moved to `XdocsTemplateParser` because it is the only class one that needs `user.dir`

Reply:
Please add required step in javadoc of test class, to make it clear for users how to run it.

Reply:
Done.